pyiron_base/project/archiving/import_archive.py

```diff
@@ -27,6 +27,8 @@ def extract_archive(archive_directory):
 
 
 def import_jobs(project_instance, archive_directory, df, compressed=True):
+    if len(set(df["job"]) & set(project_instance.job_table().job)) > 0:
+        raise ValueError("Overlapping", df["job"], project_instance.job_table().job)
```

Comment:
It seems very reasonable to me to test and fail when there is a conflict, but given my comment in the main thread I guess this check will return false-positives for overlap since `project_instance.job_table().job` can be non-unique when jobs of the same name have different (sub)projects.

I like the idea of introducing this, but I guess the tests need to be expanded to make sure we _do_ fail when we want to (direct overlap in the main project, direct overlap in any subproject) and _don't_ fail when we shouldn't (no overlap, or overlap in job column but with different sub-projects)

Reply:
nit: I've taken to introducing custom errors for most things, so that if I ever need to test it by setting it up to intentionally fail I can make sure it's failing for the exact reason I am wanting and not raising a similar error early for some other reason. E.g.

```python
class OverlappingJobsError(ValueError):
    """Raise when a project tries to import a job whose name it already has"""
```

Reply:
nit: You can pass a tuple in and the message comes out as a tuple, but I guess it would be prettier to f-string these pieces together?

Reply:
> nit: I've taken to introducing custom errors for most things, so that if I ever need to test it by setting it up to intentionally fail I can make sure it's failing for the exact reason I am wanting and not raising a similar error early for some other reason. E.g.
> 
> ```python
> class OverlappingJobsError(ValueError):
>     """Raise when a project tries to import a job whose name it already has"""
> ```

I do not have time to check. Do these errors get captured by a `except ValueError` clause?

Reply:
@niklassiemer, yes, it also counts as the parent class for the purposes of assert raises